src/test/java/org/junitpioneer/jupiter/RangeSourceProviderTests.java

```diff
@@ -0,0 +1,285 @@
+	}
+
+	private static Number displayNameToNumber(String displayName) {
+		String[] split = displayName.split(" ");
+		String type = split[0];
+		String value = split[1];
+
+		try {
+			Class<?> valueClass = Class.forName("java.lang." + type);
+			Method parseMethod = valueClass.getMethod("valueOf", String.class);
+			return (Number) parseMethod.invoke(null, value);
+		}
+		catch (Exception ignore) {
+			return null;
+		}
+	}
+
+	@Nested
+	class RangeTestCases {
+		@ParameterizedTest(name = "Byte {0}")
+		@ByteRangeSource(from = 0, to = 10)
+		public void ascendingByte(byte param) {
+		}
+
+		@ParameterizedTest(name = "Byte {0}")
+		@ByteRangeSource(from = -3, to = -7, step = -1, closed = true)
+		public void descendingByte(byte param) {
+		}
+
+		@ParameterizedTest(name = "Short {0}")
+		@ShortRangeSource(from = 130, to = 136)
+		public void ascendingShort(short param) {
+		}
+
+		@ParameterizedTest(name = "Short {0}")
+		@ShortRangeSource(from = -140, to = -144, step = -2, closed = true)
+		public void descendingShort(short param) {
+		}
+
+		@ParameterizedTest(name = "Integer {0}")
+		@IntRangeSource(from = 40_000, to = 40_005)
+		public void ascendingInt(int param) {
+		}
+
+		@ParameterizedTest(name = "Integer {0}")
+		@IntRangeSource(from = -40_000, to = -42_000, step = -2000, closed = true)
+		public void descendingInt(int param) {
+		}
+
+		@ParameterizedTest(name = "Long {0}")
+		@LongRangeSource(from = 6_000_000_000L, to = 6_000_000_003L)
+		public void ascendingLong(long param) {
+		}
+
+		@ParameterizedTest(name = "Long {0}")
+		@LongRangeSource(from = -6_000_000_000L, to = -6_000_000_400L, step = -100, closed = true)
+		public void descendingLong(long param) {
+		}
+
+		@ParameterizedTest(name = "Float {0}")
+		@FloatRangeSource(from = 2.2F, to = 5.2F)
+		public void ascendingFloat(float param) {
+		}
+
+		@ParameterizedTest(name = "Float {0}")
+		@FloatRangeSource(from = -0.3F, to = -0.6F, step = -0.1F, closed = true)
+		public void descendingFloat(float param) {
+		}
+
+		@ParameterizedTest(name = "Double {0}")
+		@DoubleRangeSource(from = 8.4, to = 10.4)
+		public void ascendingDouble(double param) {
+		}
+
+		@ParameterizedTest(name = "Double {0}")
+		@DoubleRangeSource(from = -2.0, to = -3.0, step = -1, closed = true)
+		public void descendingDouble(double param) {
+		}
+
+		@Test
+		public void noAnnotation() throws NoSuchMethodException {
+			Object dummy = new Object() {
+				@Override
+				public String toString() {
+					return "";
+				}
+			};
+
+			ExtensionContext ec = spy(ExtensionContext.class);
+			when(ec.getElement()).thenReturn(Optional.of(dummy.getClass().getMethod("toString")));
+
+			RangeSourceProvider provider = new RangeSourceProvider();
+			PreconditionViolationException e = assertThrows(PreconditionViolationException.class,
+				() -> provider.provideArguments(ec));
+			assertEquals("Expected exactly one annotaiton to provide an ArgumentSource, found 0.", e.getMessage());
+		}
```

Comment:
I think there are better ways to test misconfigurations. Shouldn't it be possible to create a regular test method that is poorly annotated and use the common `executeTestsForClass` approach to verify errors?